### DISCUSSION TITLE
How can we align text HTML element to center vertically without using `flex` and `position` CSS property ?

### DISCUSSION THREAD

**Discussion by ProPhycient5** [tailwindlabs/tailwindcss#10165] [Help] [answered]
Like we use the `text-align` CSS property to align any text HTML element in a horizontal direction. Similarly, how can I align text to center vertically without using the `flex` and `position` properties????

**Comment by wongjn** [comment, marked as answer]
You could use a grid layout, `translateY()`, `margin-top`, `padding-top`, etc: https://play.tailwindcss.com/Of3cRJylRf

**Reply by ProPhycient5** [reply]
Yeah, this works but I am looking for only one CSS property like `text-align` to align any text `horizontally`.

**Reply by wongjn** [reply]
The `translateY()`, `margin-top` and `padding-top` solutions are all one-CSS-property solutions! As an additional one-property solution, you could use `vertical-align: middle` on a sibling inline element that is the height that you'd like to align vertically: https://play.tailwindcss.com/flYR2o6mC2

**Reply by ProPhycient5** [reply]
Why is there no ending `</div>` ???

**Reply by wongjn** [reply]
Because I did not write it in.

**Reply by ProPhycient5** [reply]
Copying the code from TailwindCSS play, it shows this code, how is this correct code ?
```
<div class="h-80 border border-blue-200 m-10">
  <div class="h-full inline-block align-middle">
  </div>
Foo 

```

**Reply by wongjn** [reply]
It is *correct* code in the sense that it is parsable by the browser, otherwise I would presume whether it is "correct code" depends on the criteria one conceives for "correct code".

**Reply by ProPhycient5** [reply]
okay @wongjn I was aware of this. Thanks.
